packages/admin/src/emailCampaigns/form/TestEmailCampaignForm.tsx

```diff
@@ -1,38 +1,64 @@
-import { useApolloClient } from "@apollo/client";
-import { Field, FinalForm, FinalFormInput, SaveButton } from "@comet/admin";
+import { gql, useApolloClient, useQuery } from "@apollo/client";
+import { Field, FinalForm, FinalFormSelect, SaveButton } from "@comet/admin";
 import { Newsletter } from "@comet/admin-icons";
 import { AdminComponentPaper, AdminComponentSectionGroup } from "@comet/blocks-admin";
-import { Card, FormHelperText, Typography } from "@mui/material";
+import { Card } from "@mui/material";
 import React from "react";
 import { FormattedMessage } from "react-intl";
 
+import { GQLEmailCampaignContentScopeInput } from "../../graphql.generated";
+import { GQLBrevoTestContactsSelectListFragment } from "./TestEmailCampaignForm.generated";
 import { SendEmailCampaignToTestEmailsMutation } from "./TestEmailCampaignForm.gql";
 import { GQLSendEmailCampaignToTestEmailsMutation, GQLSendEmailCampaignToTestEmailsMutationVariables } from "./TestEmailCampaignForm.gql.generated";
 
 interface FormProps {
-    testEmails: string;
+    testEmails: string[];
 }
 
 interface TestEmailCampaignFormProps {
     id?: string;
     isSendable?: boolean;
+    scope: GQLEmailCampaignContentScopeInput;
 }
 
-export const TestEmailCampaignForm = ({ id, isSendable = false }: TestEmailCampaignFormProps) => {
+const brevoTestContactsSelectFragment = gql`
+    fragment BrevoTestContactsSelectList on BrevoContact {
+        id
+        email
+    }
+`;
+
+const brevoTestContactsSelectQuery = gql`
+    query BrevoTestContactsGridSelect($offset: Int, $limit: Int, $email: String, $scope: EmailCampaignContentScopeInput!) {
+        brevoTestContacts(offset: $offset, limit: $limit, email: $email, scope: $scope) {
+            nodes {
+                ...BrevoTestContactsSelectList
+            }
+            totalCount
+        }
+    }
+    ${brevoTestContactsSelectFragment}
+`;
+
+export const TestEmailCampaignForm = ({ id, isSendable = false, scope }: TestEmailCampaignFormProps) => {
     const client = useApolloClient();
 
-    async function submitTestEmails({ testEmails }: FormProps) {
-        const emailsArray = testEmails.trim().split("\n");
+    const { data, loading, error } = useQuery(brevoTestContactsSelectQuery, {
+        variables: { offset: 0, limit: 50, email: "", scope },
```

Comment:
Why is email set here? 

I doubt there will be more than 50 tests contacts but I guess this should be handled somehow... @raphaelblum what do you think about the limit?

Reply:
Couldn't we use Autocomplete for this?
https://storybook.comet-dxp.com/?path=/story/docs-hooks-useasyncoptionsprops--autocomplete

Reply:
I think I addressed this issue about reloading values in an async select before, I do not know if it is already possible to load more (on scrolling, input change..)?

Reply:
I removed the email: 411562681f893ca04a75037648406284d069a843

As discussed, I changed the limit to 100: 0a664dbc5831cf32b658eb55003f78532166cb21 and added a check in the api to prevent creating more than 100 contacts: 13d2f5f18e0142fe41a219923276fb01f06a70e8, 5e991c3df8d1e9b7ca17e9677c54b8238b710568

In the frontend the button to create a new contact is disabled as soon as there are 100 contacts and a tooltip gives information about that: f2403cbf7d3988cf6ac0661863f5a1ecfe317027

<img width="1477" alt="Screenshot 2024-12-13 at 10 04 13" src="https://github.com/user-attachments/assets/d86879ce-e4ed-4bfc-988d-58f3967a90b0" />